src/DynamoCoreWpf/Extensions/ViewExtensionManager.cs

```diff
@@ -14,8 +15,49 @@ internal class ViewExtensionManager : IViewExtensionManager, ILogSource
         public ViewExtensionManager()
         {
             viewExtensionLoader.MessageLogged += Log;
+            this.ExtensionLoader.ExtensionLoading += SubscribeViewExtension;
+            this.ExtensionRemoved += UnsubscribeViewExtension;
         }
 
+        private void RequestAddViewExtensionHandler(dynamic viewExtension)
+        {
+            if (viewExtension is IViewExtension)
+            {
+                this.Add(viewExtension as IViewExtension);
+            }
+
+        }
+        private IViewExtension RequestLoadViewExtensionHandler(string extensionPath)
+        {
+            // If the path is a viewExtension - load it.
+            if ((extensionPath.Contains("_ViewExtensionDefinition")))
+            {
+                return this.ExtensionLoader.Load(extensionPath);
+            }
+            return null;
+        }
+
+
+        private void UnsubscribeViewExtension(IViewExtension obj)
+        {
+            if (obj is IExtensionSource)
+            {
+                (obj as IExtensionSource).RequestLoadExtension -= RequestLoadViewExtensionHandler;
+                (obj as IExtensionSource).RequestAddExtension -= RequestAddViewExtensionHandler ;
+            }
+        }
+
+        private void SubscribeViewExtension(IViewExtension obj)
+        {
+            //if this extension could be a source of other extensions (like packageManagerExtension) then
+            //lets handle those requests.
+            if (obj is IExtensionSource)
+            {
+                (obj as IExtensionSource).RequestLoadExtension += RequestLoadViewExtensionHandler;
+                (obj as IExtensionSource).RequestAddExtension += RequestAddViewExtensionHandler;
+            }
+
+        }
```

Comment:
There's a lot of distracting white space issues in this file. Sometimes a method has an extra line, sometimes methods are separated by a new line, sometimes not. No one likes to talk about source code formatting.

It like just style, but it _does_ matter. It's harder to read the code in Github, I feel the need to comment on it (distracting from meaningful review comments), it's harder to write tooling, and ultimately can be solved with automation.

Reply:
will fix them up.